### DISCUSSION TITLE
[REQUEST] Adding branch filter to delete trigger in github actions

### DISCUSSION THREAD

**Discussion by logyball** [community/community#10589] [Actions] [answered]
Github action supports[ a delete trigger to kick off workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#delete).  It is very useful for cleaning up resources deployed via temporary deployments.  In the same way that the `push` trigger supports a branch-based filter, it would be an awesome feature if the `delete` trigger also supported filtering based on branch name.

e.g.

```yaml
on:
  delete:
    branches:
      - feature/*
```

Would only trigger actions based on deletions of branches matching the `feature/*` semantic.

Thanks for reading!

**Comment by chrispat** [comment, marked as answer]
Hi @logyball thank you for your feedback.  

I agree this would be a nice improvement.  I have added this to our internal backlog for the team to prioritize.

**Reply by jflatow** [reply]
Is this going to happen? It seems like a glaring omission of the trigger interface

**Reply by m3ac-AllbrittenJ** [reply]
just repeating others: this feature is needed for my org too

**Comment by triethuynh1408** [comment]
Please done this feature. I need it :D

**Comment by willipsantos** [comment]
I need this feature too.
this will help us a lot

**Comment by cee-money** [comment]
Yes, this would be ideal! Just wondering where on the roadmap this enhancement is? @chrispat

**Comment by markovar** [comment]
has this being done already? 🙏

**Comment by ThomasMarcelo** [comment]
I also need this! 
We have dynamic environments based on specific branch names. The workflow for destroying the infrastructure of these branches is triggered using `on: delete`. But now it triggers on every branch deletion. I have to use if statements on every step to prevent them from running for non-matching branches, and it ruins our workflow runs history.

**Comment by gberche-orange** [comment]
Also it is currently hard to discover that `on:delete` does not yet support a nested `branches` object

The json schema should reject any object within the delete 
https://github.com/SchemaStore/schemastore/blob/f0aa8573dff529d5122970500bab36859ff39f10/src/schemas/json/github-workflow.json#L1035-L1039
![image](https://user-images.githubusercontent.com/4748380/213299104-27a4d2e8-cb3d-4655-9ae4-54923a2c8664.png)
![image](https://user-images.githubusercontent.com/4748380/213299177-c3c02c01-1cd8-4164-9874-3e77c879df11.png)

**Comment by pneigel-ca** [comment]
I am using GHES 3.5.

I have worked around this by declaring an `if` condition at the beginning of the workflow:
```yml
name: Branch Deployment Cleanup Pipeline
on: delete
jobs:
  destroy:
    if: |
      github.event.ref_type =='branch' && 
      startswith(github.event.ref, '<prefix>')
```

If these two conditions are not met (misnamed branches, tags), it will result in this:

![image](https://user-images.githubusercontent.com/48532268/224178651-78e4072d-bf1f-431f-908b-e9d75eed0fc7.png)

**Comment by constgen** [comment]
But the workflow run still is present in the history. It may confuse when you delete a lot of Pull Request branches

**Reply by pneigel-ca** [reply]
This is true. It's not a replacement for the feature, but a workaround with some compromise.

**Comment by kostiuchenkodg** [comment]
I also need this feature and if-statement is not working for me. I need to trigger workflow on the deletion branch with a specific name, but it's skipped every time
```
`name: Cleanup

on:
  delete:

jobs:
  delete-product-version:
    if: github.event.ref_type == 'branch' && startsWith(github.event.ref, 'refs/heads/sc-')`
```
    
also tried this one, but it's skipped everytime too:

if: startsWith(github.event.ref, 'refs/heads/sc-')

**Reply by rockcheung2004** [reply]
`github.event.ref` in the `delete` event is just the branch name without `refs/heads/`

so in your case this may work: 
`if: github.event.ref_type == 'branch' && startsWith(github.event.ref, 'sc-')`

It is different from the `github.ref` that you get from other event payloads, which includes the `refs/heads/`.

**Reply by kostiuchenkodg** [reply]
@rockcheung2004 Thanks, you've made my day

**Comment by rockcheung2004** [comment]
I would also suggest removing the limitation where `delete` can only be triggered after the workflow is merged into the main branch's code. You cannot first test your workflow before merging into main.

**Reply by chrispat** [reply]
In the case of delete the branch is gone before the workflow runs which means we can't run the workflow from that branch. Given that we have to pick a consistent place from which to run the workflow and the default branch is the only logical option.

**Comment by Howu48** [comment]
~~Hi @chrispat . This feature is not in [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore)(and vscode's gh actions extention warns me), but it worked well....... 🤔 
has this being done already?
(I'm using GHES 3.9)~~
never mind. it was just server issue

**Comment by OlesYudin** [comment]
Hi @Howu48! Could you please share your configuration of the job? I tested this feature in the non-Enterprise version of GitHub and it did not work as expected. 
Here is my test config:
```yaml
name: TestOnDelete

on:
  delete:
    branches:
      - 'test-*'

jobs:
  test:
    runs-on: ubuntu-latest
    ...
```
Using this configuration each time I delete a branch - GitHub runs a new action, but in my case, I need to run the action only if delete a special branch, for example, "test-1".

Solution with the condition in a job works well, but it generates a lot of skipped actions.
```yaml
if: github.event.ref_type == 'branch' && startsWith(github.event.ref, 'test-')
```

**Reply by Howu48** [reply]
Hi, and Sorry. A few days after the comment, I found out it was just a temporary our GHES server issue (other actions weren't triggered either).😓 
So currently, this feature is not available.

**Reply by OlesYudin** [reply]
Thank you for your response :cry:

**Comment by OlesYudin** [comment]
Are there any updates?

**Comment by Aarbel** [comment]
also including `branches-ignore` could be a must, is this supported now ?

**Reply by fvidak01** [reply]
Doesn't seem to be supported at all. :shrug:

**Comment by AurimasNav** [comment]
Looking forward to it.

**Comment by itsyoboieltr** [comment]
I also require this feature :D

**Comment by zaclummys** [comment]
Any updates on that?

**Comment by grilla99** [comment]
Yes please

**Comment by JohnCido** [comment]
Any updates on this? It's been four years now.
